commit d3908fdcb6820fa034dd616aff51f78d1a324384 (refs/changes/52/752/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-10-20T11:49:31-07:00 (3 years ago)
    
    DOCKER-953 'docker pull' error message for a download error says "Internal error"

diff --git a/lib/backends/sdc/images.js b/lib/backends/sdc/images.js
index 54ae62d..b973aef 100644
--- a/lib/backends/sdc/images.js
+++ b/lib/backends/sdc/images.js
@@ -1429,6 +1429,11 @@ function pullImage(opts, callback) {
                     + 'to pull %s (%s)', opts.rat.index.name,
                     opts.rat.remoteName, opts.req_id)
             };
+        } else if (err.code === 'DownloadError') {
+            // E.g. OAuth auth to a Docker Registry before DOCKER-771.
+            payload = {
+                status: format('Error downloading %s: %s', imgName, err.message)
+            };
         } else if (err.code === 'NotImplemented') {
             // E.g. OAuth auth to a Docker Registry before DOCKER-771.
             payload = {
